src/ui/public/chrome/directives/kbn_chrome.js

```diff
@@ -26,15 +27,16 @@ export default function (chrome, internals) {
       },
 
       controllerAs: 'chrome',
-      controller($scope, $rootScope, $location, $http) {
+      controller($scope, $rootScope, $location, $http, Private) {
+        const unhashStates = Private(UnhashStatesProvider);
 
         // are we showing the embedded version of the chrome?
         internals.setVisibleDefault(!$location.search().embed);
 
         // listen for route changes, propogate to tabs
         const onRouteChange = function () {
           let { href } = window.location;
-          internals.trackPossibleSubUrl(href);
+          internals.trackPossibleSubUrl(unhashStates.inAbsUrl(href));
```

Comment:
I think if we add some variables here we can add context and improve readability:

```javascript
const hashedUrl = window.location.href;
const unhashedUrl = unhashStates.inAbsUrl(hashedUrl);
internals.trackPossibleSubUrl(unhashedUrl);
```